File: kakao/level2/튜플.py
def solution(st):
    answer = []
    st = st[2:-2]
    st = sorted(st.split("},{"), key=lambda s: len(s))
    visited = set()
    for s in st:
        elements = s.split(",")
        for e in elements:
            if e not in visited:
                visited.add(e)
                answer.append(int(e))
    return answer


s = "{{2},{2,1},{2,1,3},{2,1,3,4}}"
print(solution(s)) # [2, 1, 3, 4]

s = "{{1,2,3},{2,1},{1,2,4,3},{2}}"
print(solution(s)) # [2, 1, 3, 4]

s = "{{20,111},{111}}"
print(solution(s)) # [111, 20]

s = "{{123}}"
print(solution(s)) # [123]

s = "{{4,2,3},{3},{2,3,4,1},{2,3}}"
print(solution(s)) # [3, 2, 4, 1]

File: Java/programmers/kakao/level2/튜플.java
package CodeTest.Java.programmers.kakao.level2;

import java.util.*;

public class 튜플 {
    public int[] solution(String s) {
        String[] st = s.replaceAll("[{]", " ")
            .replaceAll("[}]", " ")
            .trim()
            .split(" , ");
        Arrays.sort(st, (o1,o2)->{return o1.length()-o2.length();});
        
        Set<String> visited = new HashSet<>();
        int[] answer = new int[st.length];
        
        int idx = 0;
        for (String i : st) {
            String[] elements = i.split(",");
            for (String e : elements) {
                if (!visited.contains(e)) {
                    visited.add(e);
                    answer[idx++] = Integer.parseInt(e);
                }
            }
        }

        return answer;
    }
}
